crates/oxc_linter/src/rules/import/first.rs

```diff
@@ -0,0 +1,234 @@
+use oxc_ast::{
+    ast::{Statement, TSModuleReference},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+fn first_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Import statements must come first")
+        .with_help("Move import statement to the top of the file")
+        .with_label(span)
+}
+
+fn absolute_first_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Relative imports before absolute imports are prohibited")
+        .with_help("Move absolute import above relative import")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct First {
+    absolute_first: AbsoluteFirst,
+}
+
+#[derive(Debug, Default, Clone)]
+enum AbsoluteFirst {
+    AbsoluteFirst,
+    #[default]
+    DisableAbsoluteFirst,
+}
+
+impl AbsoluteFirst {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "absolute-first" => Self::AbsoluteFirst,
+            _ => Self::DisableAbsoluteFirst,
+        }
+    }
+}
```

Comment:
should this be 

```
impl From for AbsoluteFirst...
```

https://doc.rust-lang.org/rust-by-example/conversion/from_into.html